src/main/java/io/cdap/plugin/servicenow/source/ServiceNowMultiSource.java

```diff
@@ -0,0 +1,120 @@
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package io.cdap.plugin.servicenow.source;
+
+import io.cdap.cdap.api.annotation.Description;
+import io.cdap.cdap.api.annotation.Name;
+import io.cdap.cdap.api.annotation.Plugin;
+import io.cdap.cdap.api.data.batch.Input;
+import io.cdap.cdap.api.data.format.StructuredRecord;
+import io.cdap.cdap.api.data.schema.Schema;
+import io.cdap.cdap.api.dataset.lib.KeyValue;
+import io.cdap.cdap.etl.api.Emitter;
+import io.cdap.cdap.etl.api.FailureCollector;
+import io.cdap.cdap.etl.api.PipelineConfigurer;
+import io.cdap.cdap.etl.api.StageConfigurer;
+import io.cdap.cdap.etl.api.action.SettableArguments;
+import io.cdap.cdap.etl.api.batch.BatchSource;
+import io.cdap.cdap.etl.api.batch.BatchSourceContext;
+import io.cdap.plugin.common.LineageRecorder;
+import io.cdap.plugin.common.SourceInputFormatProvider;
+import io.cdap.plugin.servicenow.source.util.ServiceNowConstants;
+import io.cdap.plugin.servicenow.source.util.ServiceNowTableInfo;
+
+import org.apache.hadoop.conf.Configuration;
+import org.apache.hadoop.io.NullWritable;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.Collection;
+import java.util.List;
+import java.util.Objects;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+/**
+ * A {@link BatchSource} that reads data from multiple tables in Service Now.
+ */
+@Plugin(type = BatchSource.PLUGIN_TYPE)
+@Name(ServiceNowConstants.PLUGIN_NAME_MULTI_SOURCE)
+@Description("Reads from multiple tables in Service Now. " +
+  "Outputs one record for each row in each table, with the table name as a record field. " +
+  "Also sets a pipeline argument for each table read, which contains the table schema. ")
+public class ServiceNowMultiSource extends BatchSource<NullWritable, StructuredRecord, StructuredRecord> {
+  private static final Logger LOG = LoggerFactory.getLogger(ServiceNowMultiSource.class);
+
+  private final ServiceNowMultiSourceConfig conf;
+
+  public ServiceNowMultiSource(ServiceNowMultiSourceConfig conf) {
+    this.conf = conf;
+  }
+
+  @Override
+  public void configurePipeline(PipelineConfigurer pipelineConfigurer) {
+    super.configurePipeline(pipelineConfigurer);
+
+    LOG.debug("Validate config during `configurePipeline` stage: {}", conf);
+    StageConfigurer stageConfigurer = pipelineConfigurer.getStageConfigurer();
+    FailureCollector collector = stageConfigurer.getFailureCollector();
+
+    conf.validate(collector);
+    collector.getOrThrowException();
+    // Since we have validated all the properties, throw an exception if there are any errors in the collector.
+    // This is to avoid adding same validation errors again in getSchema method call
+    if (conf.shouldGetSchema()) {
+      Set<ServiceNowTableInfo> tableInfos = ServiceNowMultiInputFormat.fetchTablesInfo(conf);
+      stageConfigurer.setOutputSchema(tableInfos.stream().findFirst().get().getSchema());
```

Comment:
why is output schema set to schema of the first table? I think we can just set it to null. Ex - https://github.com/data-integrations/multi-table-plugins/blob/develop/src/main/java/io/cdap/plugin/MultiTableDBSource.java#L84

Reply:
CDAP currently doesn't support displaying schema for all tables, if there is more than 1 table. So, it was mutually agreed that the schema of the first table entered by the user gets displayed as the output schema.
As per the reference shared by you for `MultiTableDBSource`, I've set the output schema to `null`. 6c4965a

I have one query here.
Since the Service Now Batch Source Plugin Reporting Mode's  behaviour is similar to Service Now Batch Multi-Source Plugin and has output schema set to the schema of the first table, should we set the output schema to display `null` there as well?